src/index.js

```diff
@@ -49,73 +50,117 @@ function createCommand(queryName, implementationName) {
-        message: inputArr,
-        consoleProps: () => consoleProps,
-      })
+      if (options.log) {
+        options._log = Cypress.log({
+          name: queryName,
+          message: inputArr,
+          consoleProps: () => consoleProps,
+        })
+      }
 
-      return cy
-        .window({log: false})
-        .then({timeout: waitOptions.timeout + 100}, thenArgs => {
-          const getValue = () => {
-            const value = commandImpl(thenArgs.document)
-            const result = Cypress.$(value)
-
-            // Overriding the selector of the jquery object because it's displayed in the long message of .should('exist') failure message
-            // Hopefully it makes it clearer, because I find the normal response of "Expected to find element '', but never found it" confusing
-            result.selector = `${queryName}(${queryArgument(args)})`
-
-            if (result.length > 0) {
-              consoleProps.yielded = result.toArray()
-            }
-
-            return result
-          }
-
-          const resolveValue = () => {
-            // retry calling "getValue" until following assertions pass or this command times out
-            return Cypress.Promise.try(getValue).then(value => {
-              return cy.verifyUpcomingAssertions(value, waitOptions, {
-                onRetry: resolveValue,
-              })
-            })
-          }
+      const getValue = () => {
+        const value = commandImpl(win.document)
+
+        const result = Cypress.$(value)
+        if (value && options._log) {
+          options._log.set('$el', result)
+        }
 
-          if (queryRegex.test(queryName)) {
-            // For get* queries, do not retry
-            return getValue()
-          }
+        // Overriding the selector of the jquery object because it's displayed in the long message of .should('exist') failure message
+        // Hopefully it makes it clearer, because I find the normal response of "Expected to find element '', but never found it" confusing
+        result.selector = getSelector()
 
-          return resolveValue().then(subject => {
-            // Remove the error that occurred because it is irrelevant now
-            if (consoleProps.error) {
-              delete consoleProps.error
-            }
+        consoleProps.elements = result.length
+        if (result.length === 1) {
+          consoleProps.yielded = result.toArray()[0];
+        } else if (result.length > 0) {
+          consoleProps.yielded = result.toArray();
+        }
 
-            return subject
-          })
+        if (result.length > 1 && !/all/i.test(queryName)) {
+          // Is this useful?
+          throw Error(`Found multiple elements with the text: ${queryArgument(args)}`)
+        }
+
+        return result
+      }
+
+      if (queryRegex.test(queryName)) {
+        // make the timeout extremely short to ensure `query*` commands pass or fail instantly
+        options.timeout = 0
+      }
```

Comment:
Question:

Should we do this? The docs mention query is synchronous. Setting a timeout of 0 is the closest Cypress can get to being synchronous while still operating like Cypress does.

Previously the code didn't do a `cy.verifyUpcomingAssertions` with a retry loop so `query*` had an inconsistent experience and assertions like `.should('not.exist')` didn't work

Reply:
Yes, I think we should do this, though I think that we should not override `options.timeout` if they're specified.

Also, I'm thinking that in a future breaking change we can remove `query*` and only support `find*`.

Reply:
I agree with removing `query*` in the future. The functionality can be done using a low timeout. Most of the time, this is not what users want, but they can accomplish the same thing with built-in Cypress commands:

```ts
cy.findByText('My Button', { timeout: 0 }) // won't retry because the timeout is 0, but it is not asynchronous
cy.contains('My Button', { timeout: 0 }) // consistent functionality to the above line in terms of timing
```

Reply:
I'll remove the timeout override. For `query*` it will default to `0`, but allow user override.